### PR TITLE
[IMP] sale: Hide downpayment product from user.

### PR DESCRIPTION
**Pull request by jbw-odoo**
In the settings, the Downpayments Products setting for Sales/Invoicing has become obsolete.
This commit removes the DownPayment Product from the Invoicing Settings. And hide the DownPayment Product from the user by archiving it and make it unclickable in SO's and invoices.

See: https://github.com/odoo/upgrade/pull/4914

Task: 3422811

### CODE REVIEW

File: addons/sale/wizard/sale_make_invoice_advance.py [hunk 1]
```diff
@@ -200,6 +195,7 @@ def _create_invoices(self, sale_orders):
                 self.company_id.sale_down_payment_product_id = self.env['product.product'].create(
                     self._prepare_down_payment_product_values()
                 )
+                self.company_id.sale_down_payment_product_id.product_tmpl_id.active = False  # hide down payment product from user
```
Comment (qdp-odoo) [inline addons/sale/wizard/sale_make_invoice_advance.py:198]: already done in _prepare_down_payment_product_values()

File: addons/sale/wizard/sale_make_invoice_advance.py [hunk 2]
```diff
@@ -250,7 +246,8 @@ def _prepare_down_payment_product_values(self):
             'invoice_policy': 'order',
             'company_id': self.company_id.id,
             'property_account_income_id': self.deposit_account_id.id,
-            'taxes_id': [Command.set(self.deposit_taxes_id.ids)],
+            'taxes_id': False,
+            'active': False,  # hide down payment product from user
```
Comment (qdp-odoo) [inline addons/sale/wizard/sale_make_invoice_advance.py:250]: why create it, and then archive it?

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
I don't get the philosophy


if the purpose is to have an archived downpayment product, you probably need to add active_test here and there to avoid creating a new downpayment product at each downpayment.

File: addons/sale/wizard/sale_make_invoice_advance.py (hunk 1 above)
Comment (jbw-odoo) [reply addons/sale/wizard/sale_make_invoice_advance.py:198]: > already done in _prepare_down_payment_product_values()

The product.product is set to active false, but the product.template is not at that point

File: addons/sale/wizard/sale_make_invoice_advance.py (hunk 2 above)
Comment (jbw-odoo) [reply addons/sale/wizard/sale_make_invoice_advance.py:250]: It's archived to not show up when the user interacts with the UI, but it is still used as down payment product.

**Comment by jbw-odoo** [comment]
> I don't get the philosophy
> 
> if the purpose is to have an archived downpayment product, you probably need to add active_test here and there to avoid creating a new downpayment product at each downpayment.

It'd archived to be hidden from the user.
It is only created once and set on the company as  `sale_down_payment_product_id`. And then used as down payment product for the company even if archived.
(the same way OdooBot user is archived to be hidden but still used as automated user)

**Comment by tsb-odoo** [comment]
@jbw-odoo 
I created a new task to separate things properly.
I also edited the PR message that pretended changes were happening regarding the application of fiscal positions, which isn't covered yet by this PR. The PR message refers to the proper task.
Can you please check the commit messages that also need to be corrected ?
Apart from that, LGTM functionally speaking.

File: addons/sale/static/src/js/sale_product_field.js [hunk 3]
```diff
@@ -39,9 +39,10 @@ export class SaleOrderLineProductField extends Many2OneField {
         // product form should be accessible if the widget field is readonly
         // or if the line cannot be edited (e.g. locked SO)
         return (
-            this.props.readonlyField ||
-            (this.props.record.model.root.activeFields.order_line &&
-                this.props.record.model.root._isReadonly("order_line"))
+            this.props.context?.is_downpayment !== true && ( this.props.readonlyField || (
```
Comment (Feyensv) [inline addons/sale/static/src/js/sale_product_field.js:42]: Can't you access the field in the record data instead of relying on the context ?

**Review by Feyensv — COMMENTED** [review]
> Remove the DownPayment Product from the Invoicing Settings since it's completely ignored by the code.

Well, it is used by the code :thinking:

I don't really get the purpose/idea of this task.
We're adding some strange magic complexifying the code, to only hide a bit the dp product (and remove a setting) ?

File: addons/sale/static/src/js/sale_product_field.js (hunk 3 above)
Comment (jbw-odoo) [reply addons/sale/static/src/js/sale_product_field.js]: I can indeed. thx.

**Comment by jbw-odoo** [comment]
Task is axed in it's current form.

**Comment by jbw-odoo** [comment]
> axed? like.. split in two, by an axe ?

image

**Pull request closed without merging**